flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskExecutor.java

```diff
@@ -932,9 +932,9 @@ public void heartbeatFromResourceManager(ResourceID resourceID) {
         final Task task = taskSlotTable.getTask(executionAttemptID);
 
         if (task != null) {
-            task.triggerCheckpointBarrier(checkpointId, checkpointTimestamp, checkpointOptions);
-
-            return CompletableFuture.completedFuture(Acknowledge.get());
+            return task.triggerCheckpointBarrier(
+                            checkpointId, checkpointTimestamp, checkpointOptions)
+                    .thenApply(result -> Acknowledge.get());
```

Comment:
I think this future should be completed upon trigger succeeds, i.e. immediately after `task.triggerCheckpointBarrier` returns. From `TaskExecutorGateway`:
```
     * @return Future acknowledge if the checkpoint has been successfully triggered
```

With the current change, it will be completed upon sync phase completion.

Given the above and that the result is still ignored in `RpcTaskManagerGateway`, I'm a bit skeptical of returning the future through the call chain.

WDYT?

Reply:
Right, I am also not sure about the change. However if we go with ignoring the result (option 2. from the description) we should revisit the whole stack, as there is a somewhat bigger logic in `org.apache.flink.streaming.runtime.tasks.MultipleInputStreamTask#pendingCheckpointCompletedFutures` which does not work.

Reply:
> Given the above and that the result is still ignored in RpcTaskManagerGateway, I'm a bit skeptical of returning the future through the call chain.

If the result is respected in that side, then the whole call chain will fail due to rpc timeout(defaults to 10s). I guess it is not desired as it will interfere with checkpoint timeout.

Reply:
I agree, with both points.

And I think it's fine to drop that logic in `MultipleInputStreamTask`. WDYT @pnowojski ?

Reply:
about `MultipleInputStreamTask#pendingCheckpointCompletedFutures` that's a shitty code that I had to add/keep for testing purposes if I recall correctly :/ But maybe I've missed something, so feel free to re-investigate this if it's causing troubles.

Reply:
Right, I'll revisit that logic then and I will drop the result type of `triggerCheckpointAsync`